File: ios/MutualOffline/Models/Device.swift
import Foundation

struct Device: Codable, Identifiable, Hashable {
    let id: Int
    let userId: Int
    let deviceUuid: String
    let platform: String
    let deviceName: String?
    let appVersion: String?
    let pushToken: String?
    let isActive: Bool
    let lastSeenAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case deviceUuid = "device_uuid"
        case platform
        case deviceName = "device_name"
        case appVersion = "app_version"
        case pushToken = "push_token"
        case isActive = "is_active"
        case lastSeenAt = "last_seen_at"
    }
}

File: backend/app/Models/Device.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Device extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'device_uuid',
        'platform',
        'device_name',
        'app_version',
        'push_token',
        'is_active',
        'revoked_at',
        'last_seen_at',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'revoked_at' => 'datetime',
            'last_seen_at' => 'datetime',
        ];
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }
}
